gym/envs/phys2d/bipedal_walker.py

```diff
@@ -0,0 +1,731 @@
+            HIP_OFFSET_X,
+            HULL_Z,
+            -HIP_OFFSET_X,
+            THIGH_Z,
+            THIGH_Z,
+            LEG_Z,
+            THIGH_Z,
+            LEG_Z,
+            STRENGTH_HIP,
+            STRENGTH_KNEE,
+            STRENGTH_HIP,
+            STRENGTH_KNEE,
+            INITIAL_X,
+            TERRAIN_SIZE / 2,
+        )
+
+        self.sys = brax.System(text_format.Parse(_SYSTEM_CONFIG, brax.Config()))
+
+        self.body_sizes = [
+            self._get_2d_box_size(body) for body in self.sys.config.bodies
+        ]
+
+        self.brax_step = jax.jit(self.sys.step)
+        self.qp = self.sys.default_qp()
+
+        self.qp, info = self.brax_step(self.qp, jp.array([0, 0, 0, 0]))
+        self.state = self._get_state(info)
+        self.renderer.render_step()
+        if not return_info:
+            return self.state
+        else:
+            return self.state, {}
+
+    def step(self, action):
+        action = jp.clip(action, -1, +1).astype(jp.float32)
+        self.qp, info = self.brax_step(
+            self.qp,
+            jp.array(
+                [
+                    action[LEFT_HIP],
+                    action[LEFT_KNEE],
+                    action[RIGHT_HIP],
+                    action[RIGHT_KNEE],
+                ]
+            ),
+        )
+        self.state = self._get_state(info)
+        self.renderer.render_step()
+
+        shaping = (
+            3.5 * self.qp.pos[HULL][0]
+        )  # moving forward is a way to receive reward (normalized to get 300 on completion)
+        shaping -= 5.0 * abs(
+            self.state[0]
+        )  # keep head straight, other than that and falling, any behavior is unpunished
+
+        reward = 0
+        if self.prev_shaping is not None:
+            reward = shaping - self.prev_shaping
+        self.prev_shaping = shaping
+
+        terminated = False
+        if self.qp.pos[HULL][0] < -INITIAL_X / 2 or info.contact.vel[HULL, 2] != 0:
+            reward = -100
+            terminated = True
+        if self.qp.pos[HULL][0] > TERRAIN_SIZE - INITIAL_X:
+            terminated = True
+
+        for a in action:
+            reward -= 0.028 * np.clip(np.abs(a), 0, 1)
+            # normalized to about -50.0 using heuristic, more optimal agent should spend less
+
+        return self.state, reward, terminated, False, {}
+
+    def _get_vertices(self, x, y, w, h, rot):
+        transformed_w, transformed_h = w * SCALE, h * SCALE
+        angle = self._get_angle(rot)
+        R = jp.array([[jp.cos(angle), jp.sin(angle)], [-jp.sin(angle), jp.cos(angle)]])
+        centered_verts = jp.array(
+            [
+                transformed_w / 2 * jp.array([-1, 1, 1, -1]),
+                transformed_h / 2 * jp.array([-1, -1, 1, 1]),
+            ]
+        )
+        rot_centered_verts = (R @ centered_verts).T
+        verts = rot_centered_verts + jp.array([x, y]) * SCALE
+        return verts
+
+    def render(self, mode="human"):
```

Comment:
Personally, as the phys2d are about performance and vectorisation then I don't think we should include the rendering code for now. Possibly when the jax graphics module is release, we could replace with that